connector/src/main/java/org/geysermc/connector/entity/Entity.java

```diff
@@ -244,6 +253,31 @@ public void updateRotation(GeyserSession session, float yaw, float pitch, boolea
         updatePositionAndRotation(session, 0, 0, 0, yaw, pitch, isOnGround);
     }
 
+    /**
+     * Unlike fishing rod pulling behavior that is calculated serverside on Bedrock and clientside on Java,
+     * Bedrock player "pushing" is nonexistent on Bedrock.
+     * <br>
+     * This code checks to see if a Geyser player should be moved, and if so applies motion to the player
+     * which sends movement packets back.
+     * <br>
+     * This should be called any time an entity moves position.
+     * @param session the session of the Bedrock player
+     */
+    public void checkPlayerCollision(GeyserSession session) {
+        double distanceX = session.getPlayerEntity().getPosition().getX() - this.position.getX();
+        double distanceZ = session.getPlayerEntity().getPosition().getZ() - this.position.getZ();
+        double largestDistance = Math.max(Math.abs(distanceX), Math.abs(distanceZ));
+        if (largestDistance <= 0.65) {
+            double velocityX = (distanceX / 20) / Math.sqrt(largestDistance);
+            double velocityZ = (distanceZ / 20) / Math.sqrt(largestDistance);
```

Comment:
Can we avoid running this square root operation? This isn't the best on performance (especially on every movement), and it's much faster to square `distanceX` and `distanceZ` and compare it that way.